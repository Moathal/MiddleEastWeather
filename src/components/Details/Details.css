.infoList {
  display: flex;
  flex-direction: column;
  align-items: center;
  list-style: none;
  padding: 2px;
  width: calc(100% - 16px);
  margin: 1px;
}

.listItem {
  display: flex;
  width: calc(100% - 2px);
  justify-content: space-evenly;
  align-items: center;
  padding: 4px;
  border-color: lightgray;
  border-style: solid;
  border-width: 1px;
  margin-bottom: 10px;
  background-color: transparent;
  background-color: rgb(66, 105, 179);
  box-shadow: 0 0 6px rgb(0, 29, 56);
  border: none;
}

.info-key {
  color: rgb(255, 255, 255);
}

.info-value {
  color: lightgray;
}

small {
  color: rgb(237, 76, 139);
}
