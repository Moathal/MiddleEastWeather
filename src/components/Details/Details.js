import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { countryThunk } from '../../Redux/details';
import './Details.css';

function Details() {
  const location = useLocation();
  const info = useSelector((state) => state.countryInfo.data);
  const dispatch = useDispatch();
  const id = location.pathname.substring(9, location.pathname.length);
  useEffect(() => {
    dispatch(countryThunk(id));
  }, [dispatch]);
  return (
    <>
      {typeof (info) !== 'undefined' ? (
        <ul className="infoList">
          <li className="listItem">
            <h4 className="info-key">Country</h4>
            <p className="info-value">{info.name}</p>
          </li>
          <li className="listItem">
            <h4 className="info-key">Humidity</h4>
            <p className="info-value">
              {info.main.humidity}
              <small>%</small>
            </p>
          </li>
          <li className="listItem">
            <h4 className="info-key">Description</h4>
            <p className="info-value">{info.weather[0].description}</p>
          </li>
          <li className="listItem">
            <h4 className="info-key">
              Temperature
            </h4>
            <p className="info-value">
              {Math.floor(info.main.feels_like) - 273}
              <small> Celsius</small>
            </p>
          </li>
          <li className="listItem">
            <h4 className="info-key">Wind Speed</h4>
            <p className="info-value">
              {info.wind.speed}
              <small> meter/sec</small>
            </p>
          </li>
        </ul>
      ) : <h2 className="info-value">loading...</h2>}
    </>
  );
}

export default Details;
