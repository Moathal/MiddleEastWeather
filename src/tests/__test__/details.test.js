import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../../redux/configureStore';
import Details from '../../components/Details/Details';

it('renders Details correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
