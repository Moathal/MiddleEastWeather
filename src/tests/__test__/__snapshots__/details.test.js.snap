// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details correctly 1`] = `
<h2
  className="info-value"
>
  loading...
</h2>
`;
